daffodil-core/src/main/scala/org/apache/daffodil/runtime1/VariableRuntime1Mixin.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.runtime1
+
+import org.apache.daffodil.dsom.DFDLDefineVariable
+import org.apache.daffodil.dsom.DFDLNewVariableInstance
+import org.apache.daffodil.dsom.DFDLSetVariable
+import org.apache.daffodil.dsom.ExpressionCompilers
+import org.apache.daffodil.dsom.VariableReference
+import org.apache.daffodil.processors.VariableRuntimeData
+import org.apache.daffodil.schema.annotation.props.Found
+import org.apache.daffodil.util.Delay
+import org.apache.daffodil.util.Maybe
+import org.apache.daffodil.xml.GlobalQName
+import org.apache.daffodil.xml.XMLUtils
+
+trait DFDLDefineVariableRuntime1Mixin { self: DFDLDefineVariable =>
+
+  requiredEvaluationsAlways(maybeDefaultValueExpr)
+
+  final lazy val variableRuntimeData = {
+    val vrd = new VariableRuntimeData(
+      this.schemaFileLocation,
+      this.diagnosticDebugName,
+      this.path,
+      this.namespaces,
+      this.external,
+      this.direction,
+      Delay(maybeDefaultValueExpr),
```

Comment:
Simliar question to above, what causes this Delay to be evaluated? In this case, we do require that maybeDefaultalueExpr is evaluated, but this Delay contains the pass-by-name which isn't evaluated. Will there be any issues with this? I wonder if we need something like
```scala
lazy val maybeDefaultValueExprDelay = Delay { ... }

final lazy val variableRuntimeData  = {
  val vrd = new VariableRuntimeData(
    ...,
    maybeDefaultValueExprDelay,
    ...)
}

requiredEvaluationsAlways(maybeDefaultValueExprDelay.value)
```

Reply:
Good point. What we want it to allow construction of these cyclic objects, but something subsequently should force the delay on the outside of the runtimeData objects. Right now the required-evaluation forces the argument of the Delay, not the Delay itself which would be evaluated at serialization. No need for that. In fact per changes below, we can try to make that an error.

Reply:
Ended up with all objects that take delays (excepting the Include/Import stuff) due to cyclic connections all have an initialize member (lazy val) which is responsible for forcing the delays. So instead of having a requiredEvaluation for each delay, you have one for each object having delayed parameters.